tt_metal/impl/dispatch/kernels/command_queue_producer.hpp

```diff
@@ -16,6 +17,27 @@ inline __attribute__((always_inline)) volatile uint32_t* get_cq_issue_write_ptr(
     return reinterpret_cast<volatile uint32_t*>(CQ_ISSUE_WRITE_PTR);
 }
 
+FORCE_INLINE
+void wait_consumer_idle(volatile tt_l1_ptr uint32_t* db_semaphore_addr) {
+    while (*db_semaphore_addr != 2);
```

Comment:
can this 2 be updated to make it obvious this is the number of cmd slots on consumer side?